### PR TITLE
fix(object-search): bad performance when syncing table or view columns

### LINKED ISSUE #2519: [Bug]: Oceanbase database global search, unable to find columns
### ODC version

odc430

### OB version

*

### What happened?

image
image

### What did you expect to happen?

Search for columns

### How can we reproduce it (as minimally and precisely as possible)?

Select Oceanbase database for search

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by smallsheeeep**
#### What type of PR is this?
type-bug
module-object search

#### What this PR does / why we need it:
This PR refine the performance of syncing table/view columns, and avoid too big transaction.
1. Add `Map<String, List<DBTableColumn>> listBasicColumnsInfo(String schemaName)` in `DBSchemaAccessor` to get column using a more effective SQL.
2. Refine the transaction of `batchCreate` method in Repository class to avoiding occurs too big transaction.

#### Which issue(s) this PR fixes:

Fixes #2519 

#### Special notes for your reviewer:
Self-test passed.

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yhilmare** [comment]
you may notice this pr #2231, @PeachThinking add a new yaml named `oboracle_4_1_x.yaml`, you may wait for this pr's merging and add your new sql query to this new yaml file

**Review by yizhouxw — DISMISSED** [review]
lgtm

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/DBSchemaAccessor.java [hunk 1]
```diff
@@ -198,6 +198,12 @@ default List<String> showTables(String schemaName) {
      */
     List<DBTableColumn> listBasicViewColumns(String schemaName, String viewName);
 
+    /**
+     * Get all table and view columns(hold only basic info: schema, table and column name) in the
+     * specified schema
+     */
+    Map<String, List<DBTableColumn>> listBasicColumns(String schemaName);
```
Comment (yhilmare) [inline libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/DBSchemaAccessor.java:205]: name suggestion: listBasicTableColumnsInfo

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBColumnRepository.java [hunk 2]
```diff
@@ -66,7 +68,12 @@ default List<DBColumnEntity> batchCreate(List<DBColumnEntity> entities) {
                 .add(DBColumnEntity::getObjectId)
                 .add(DBColumnEntity::getOrganizationId)
                 .build();
-        return batchCreate(entities, sql, getter, DBColumnEntity::setId, 200);
+        Iterable<List<DBColumnEntity>> partitions = Iterables.partition(entities, batchSize);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBColumnRepository.java:71]: `DBSchemaAccessorUtil#partitionFind` will help you do this

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBObjectRepository.java [hunk 3]
```diff
@@ -61,7 +65,12 @@ default List<DBObjectEntity> batchCreate(List<DBObjectEntity> entities) {
                 .add(e -> e.getType().name())
                 .add(DBObjectEntity::getOrganizationId)
                 .build();
-        return batchCreate(entities, sql, getter, DBObjectEntity::setId, 200);
+        Iterable<List<DBObjectEntity>> partitions = Iterables.partition(entities, batchSize);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBObjectRepository.java:68]: duplicated code~ same as below

File: server/plugins/schema-plugin-api/src/main/java/com/oceanbase/odc/plugin/schema/api/ColumnExtensionPoint.java [hunk 4]
```diff
@@ -45,4 +45,12 @@ public interface ColumnExtensionPoint extends ExtensionPoint {
      */
     Map<String, List<DBTableColumn>> listBasicViewColumns(Connection connection, String schemaName);
 
+    /**
+     * List all table and view columns of the specified schema
+     *
+     * @param schemaName schema name
+     * @return view columns, only basic information (schemaName, viewName, name) is included
+     */
+    Map<String, List<DBTableColumn>> listBasicColumns(Connection connection, String schemaName);
```
Comment (yhilmare) [inline server/plugins/schema-plugin-api/src/main/java/com/oceanbase/odc/plugin/schema/api/ColumnExtensionPoint.java:54]: name suggestion: listBasicTableColumnsInfo

File: libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/DBSchemaAccessor.java (hunk 1 above)
Comment (smallsheeeep) [reply libs/db-browser/src/main/java/com/oceanbase/tools/dbbrowser/schema/DBSchemaAccessor.java]: Okay, I renamed it to `listBasicColumnsInfo`, this method will return both view and table columns.

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBColumnRepository.java (hunk 2 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBColumnRepository.java]: Good reminder, I refined it.

File: server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBObjectRepository.java (hunk 3 above)
Comment (smallsheeeep) [reply server/odc-service/src/main/java/com/oceanbase/odc/metadb/dbobject/DBObjectRepository.java]: I modified it.

File: server/plugins/schema-plugin-api/src/main/java/com/oceanbase/odc/plugin/schema/api/ColumnExtensionPoint.java (hunk 4 above)
Comment (smallsheeeep) [reply server/plugins/schema-plugin-api/src/main/java/com/oceanbase/odc/plugin/schema/api/ColumnExtensionPoint.java]: Renamed to `listBasicColumnsInfo`.

**Review by yizhouxw — APPROVED** [review]

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by PeachThinking — APPROVED** [review]
lgtm

**Pull request merged**